src/commands/force/source/manifest/create.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as os from 'os';
+import { join } from 'path';
+import { flags, FlagsConfig } from '@salesforce/command';
+import { fs } from '@salesforce/core';
+import { Messages } from '@salesforce/core';
+import { SourceCommand } from '../../../../sourceCommand';
+import { ComponentSetBuilder } from '../../../../componentSetBuilder';
+
+Messages.importMessagesDirectory(__dirname);
+const deployMessages = Messages.loadMessages('@salesforce/plugin-source', 'deploy');
+const messages = Messages.loadMessages('@salesforce/plugin-source', 'create');
+
+const manifestTypes: Record<string, string> = {
+  pre: 'destructiveChangesPre.xml',
+  post: 'destructiveChangesPost.xml',
+  destroy: 'destructiveChanges.xml',
+  package: 'package.xml',
+};
+
+interface CreateResult {
+  name: string;
+  path: string;
+}
+
+export class create extends SourceCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('examples').split(os.EOL);
+  public static readonly requiresProject = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    metadata: flags.array({
+      char: 'm',
+      description: deployMessages.getMessage('flags.metadata'),
+      exclusive: ['sourcepath'],
+    }),
+    sourcepath: flags.array({
+      char: 'p',
+      description: deployMessages.getMessage('flags.sourcePath'),
+      exclusive: ['metadata'],
+    }),
+    manifestname: flags.string({
+      char: 'n',
+      description: messages.getMessage('flags.manifestname'),
+      exclusive: ['manifesttype'],
+    }),
+    manifesttype: flags.enum({
+      description: messages.getMessage('flags.manifesttype'),
+      options: Object.keys(manifestTypes),
+      char: 't',
+    }),
+    outputdir: flags.string({
+      char: 'o',
+      description: messages.getMessage('flags.outputdir'),
+    }),
+  };
+  public requiredFlags = ['metadata', 'sourcepath'];
```

Comment:
I think a better name for this would be xorFlags, meaning 1 and only 1 of them must be set when running the command.  I think this should really move into oclif at some point too.